docs/reference/query-dsl/percolator-query.asciidoc

```diff
@@ -0,0 +1,426 @@
+
+[float]
+=== Indexing Percolator Queries
+
+Percolate queries are stored as documents in a specific format and in an arbitrary index under a reserved type with the
+name `.percolator`. The query itself is placed as is in a JSON object under the top level field `query`.
+
+[source,js]
+--------------------------------------------------
+{
+    "query" : {
+		"match" : {
+			"field" : "value"
+		}
+	}
+}
+--------------------------------------------------
+
+Since this is just an ordinary document, any field can be added to this document. This can be useful later on to only
+percolate documents by specific queries.
+
+[source,js]
+--------------------------------------------------
+{
+	"query" : {
+		"match" : {
+			"field" : "value"
+		}
+	},
+	"priority" : "high"
+}
+--------------------------------------------------
+
+Just as with any other type, the `.percolator` type has a mapping, which you can configure via the mappings APIs.
+The default percolate mapping doesn't index the query field, only stores it.
+
+Because `.percolate` is a type it also has a mapping. By default the following mapping is active:
+
+[source,js]
+--------------------------------------------------
+{
+	".percolator" : {
+		"properties" : {
+			"query" : {
+				"type" : "percolator"
+			}
+		}
+	}
+}
+--------------------------------------------------
+
+If needed, this mapping can be modified with the update mapping API.
+
+In order to un-register a percolate query the delete API can be used. So if the previous added query needs to be deleted
+the following delete requests needs to be executed:
+
+[source,js]
+--------------------------------------------------
+curl -XDELETE localhost:9200/my-index/.percolator/1
+--------------------------------------------------
+
+[float]
+==== Parameters
+
+The following parameters are required when percolating a document:
+
+[horizontal]
+`document_type`:: The type / mapping of the document being percolated. This is parameter is always required.
+`document`:: The source of the document being percolated.
+
+Instead of specifying a the source of the document being percolated, the source can also be retrieved from an already
+stored document. The `percolator` query will then internally execute a get request to fetch that document.
+
+In that case the `document` parameter can be substituted with the following parameters:
+
+[horizontal]
+`indexed_document.index`:: The index the document resides in. This is a required parameter.
+`indexed_document.type`:: The type of the document to fetch. This is a required parameter.
+`indexed_document.id`:: The id of the document to fetch. This is a required parameter.
+`indexed_document.routing`:: Optionally, routing to be used to fetch document to percolate.
+`indexed_document.preference`:: Optionally, preference to be used to fetch document to percolate.
+`indexed_document.version`:: Optionally, the expected version of the document to be fetched.
+
```

Comment:
Do we need the `indexed_document` level?  Can't we just do this for an inline document:

    GET _search
    {
      "query": {
        "percolator": {
          "type": "my_type",
          "document": {...}
        }
      }
    }

and this for an indexed document:

    GET _search
    {
      "query": {
        "percolator": {
          "index": "my_index",
          "type": "my_type",
          "id": "some_id"
        }
      }
    }

This would be similar to the terms lookup query: https://www.elastic.co/guide/en/elasticsearch/reference/2.2/query-dsl-terms-query.html#query-dsl-terms-lookup